src/dvr/dvr_rec.c

```diff
@@ -607,7 +611,7 @@ pvr_generate_filename(dvr_entry_t *de, const streaming_start_t *ss)
       break;
     *(dirsep - 1) = '\0';
     if (*x) {
-      s = cleanup_filename(cfg, x);
+      s = cleanup_filename(cfg, x, de->de_directory == NULL);
```

Comment:
According to the help, specifying the de_directory overrides all other directory settings, so there isnt any other directory component.